site/reference/live/apply/README.md

```diff
@@ -1,298 +1,36 @@
-the kpt library to distinguish between resources that do not have any
-conditions set because they are fully reconciled, from resources that have no
-conditions set because they have just been created.
-
-An example of a resource where the latest change has been observed by
-the controller which is currently in the process of reconciling would be:
-
-```yaml
-apiVersion: example.com
-kind: Foo
-metadata:
-  generation: 12
-  name: bar
-spec:
-  replicas: 1
-status:
-  observedGeneration: 12
-  conditions:
-  - lastTransitionTime: "2020-03-25T21:20:38Z"
-    lastUpdateTime: "2020-03-25T21:20:38Z"
-    message: Resource is reconciling
-    reason: Reconciling
-    status: "True"
-    type: Reconciling
-  - lastTransitionTime: "2020-03-25T21:20:27Z"
-    lastUpdateTime: "2020-03-25T21:20:39Z"
-    status: "False"
-    type: Stalled
-```
-
-The status for this resource state will be InProgress. So if the
-`--reconcile-timeout` flag is set, kpt live apply will wait until
-the `Reconciling` condition is `False` before pruning and exiting.
-
-### Examples
-
-{{% hide %}}
-
-<!-- @makeWorkplace @verifyExamples-->
-```
-# Set up workspace for the test.
-TEST_HOME=$(mktemp -d)
-cd $TEST_HOME
-```
-
-<!-- @fetchPackage @verifyExamples-->
-```shell
-export SRC_REPO=https://github.com/GoogleContainerTools/kpt.git
-kpt pkg get $SRC_REPO/package-examples/helloworld-set@next my-dir
-```
-
-<!-- @createKindCluster @verifyExamples-->
-```
-kind delete cluster && kind create cluster
-```
-
-<!-- @installResourceGroup @verifyExamples-->
-```
-kpt live install-resource-group
-```
-
-<!-- @initCluster @verifyExamples-->
-```
-kpt live init my-dir
-```
-
-{{% /hide %}}
-
-<!--mdtogo:Examples-->
-<!-- @liveApply @verifyExamples-->
-```shell
-# apply resources and prune
-kpt live apply my-dir/
-```
-
-<!-- @liveApply @verifyExamples-->
-```shell
-# apply resources and wait for all the resources to be reconciled before pruning
-kpt live apply --reconcile-timeout=15m my-dir/
-```
-
-<!-- @liveApply @verifyExamples-->
-```shell
-# apply resources and specify how often to poll the cluster for resource status
-kpt live apply --reconcile-timeout=15m --poll-period=5s my-dir/
-```
-<!--mdtogo-->
-
 ### Synopsis
 <!--mdtogo:Long-->
 ```
-kpt live apply DIR [flags]
+kpt live apply [DIR|-] [flags]
 ```
 
 #### Args
-
 ```
-DIR:
-  Path to a package directory.  The directory must contain exactly
-  one ConfigMap with the inventory object annotation.
+DIR|-:
+  Path to a directory containing KRM resources and a Kptfile with inventory
+  information. The path can be relative or absolute. Providing '-' instead
+  of the path to a directory cause kpt to read from stdin.
 ```
 
 #### Flags
-
 ```
 --poll-period:
   The frequency with which the cluster will be polled to determine
-  the status of the applied resources. The default value is every 2 seconds.
+  the status of the applied resources. The default value is 2 seconds.
 
 --reconcile-timeout:
```

Comment:
Currently, all usage in the book uses `--reconcile-timeout=15m`. Should the default for this flag be changed?

Reply:
So we could use another default value here, the challenge is that the correct value varies widely depending on what is being applied. For regular k8s resources like Deployments, a minute or two should be sufficient in almost all cases unless it involves a large number of replicas. But for resources that represent out-of-cluster resources, like Ingress or KCC, the appropriate timeout would be far more than that. The requirement for users to use `--reconcile-timeout=0` to avoid waiting also feels a little bit awkward.
If we wanted to chose another default, something like 5m would probably be appropriate.